### ISSUE TITLE
way to integrate SDK with other chains in local testing

### ISSUE THREAD

**Issue by turadg** [Agoric/Agoric#8895]
As a Security Auditor and Orc API developer, I need a way to test integration of SDK (with latest vat-ibc, vat-network, vat-transfer, and network-proposal) and other chains.

* issue to launch chain with these vats: https://github.com/Agoric/agoric-sdk/issues/8903
* Need to make sure we have the **ICA Controller** enabled in our agoric image

**Comment by 0xpatrickdev** [comment]
Regarding bootstrap tests, where go/cosmos-sdk is mocked:

I prepared some sequence diagrams based on the [current ICS-27 spec](https://github.com/cosmos/ibc/blob/9ffb1d26d3018b6efda546189ec7e43d56d23da3/spec/app/ics-027-interchain-accounts/README.md), attached below. They seem quite a bit involved at first glance, but the good news is that it seems we only need to mock ~1 step from each.  _For each Controller Chain call, we just need to mock the Relayer response._  This boils down to: 

  - IBC Connection Creation: on `ConnOpenInit`, return `ConnOpenAck`
  - ICA Channel Creation: on `ChannelOpenInit`, return `ChannelOpenAck`
  - ICA Transaction: on `SendPacket`, return `onAcknowledgePacket`
  - ICA Channel Reactivation: on `ChannelOpenInit`, return `ChannelOpenAck`
     - testing should verify `SetActiveChannelID` flow on CC side


### IBC Connection Creation

_Prerequisite to creating a channel._

```mermaid
sequenceDiagram
    participant CC as Chain A
    participant R as Relayer
    participant HC as Chain B

    CC->>R: ConnOpenInit(ClientId, CounterpartyClientId, Version)
    R->>HC: ConnOpenTry(ClientId, CounterpartyClientId, Version, CounterpartyVersion)
    HC-->>R: ConnOpenTry(Version)
    R->>CC: ConnOpenAck(ConnectionId, Version)
    CC-->>R: ConnOpenAck
    R->>HC: ConnOpenConfirm(ConnectionId)
```

### ICA Channel Creation

```mermaid
sequenceDiagram
    participant CC as Controller Chain
    participant R as Relayer
    participant HC as Host Chain

    CC->>CC: RegisterInterchainAccount()
    CC->>R: ChannelOpenInit(Order, ConnectionHops, PortIdentifier, ChannelIdentifier, Version)
    R->>HC: ChannelOpenTry(Order, ConnectionHops, PortIdentifier, ChannelIdentifier, Version)
    HC->>HC: RegisterInterchainAccount(CounterpartyPortIdentifier)
    HC-->>R: ChannelOpenTry(Version)
    R->>CC: ChannelOpenAck(PortIdentifier, ChannelIdentifier, CounterpartyVersion)
    CC->>CC: SetInterchainAccountAddress(PortID, Address)
    CC->>CC: SetActiveChannelID(PortIdentifier, ConnectionID, ChannelIdentifier)
    CC-->>R: ChannelOpenAck
    R->>HC: ChannelOpenConfirm(PortIdentifier, ChannelIdentifier)
    HC->>HC: SetActiveChannelID(CounterpartyPortIdentifier, ConnectionID, ChannelIdentifier)
```

### ICA Transaction

```mermaid
sequenceDiagram
    participant CC as Controller Chain
    participant R as Relayer
    participant HC as Host Chain

    CC->>R: SendPacket(PacketData)
    R->>HC: onRecvPacket(Packet)
    HC->>HC: Deserialize and validate packet data
    alt Successful deserialization and validation
        HC->>HC: AuthenticateTx(msgs)
        HC->>HC: ExecuteTx(msgs)
        HC-->>R: Acknowledgement(result)
    else Error
        HC-->>R: ErrorAcknowledgement(error)
    end
    R->>CC: onAcknowledgePacket(Packet, Acknowledgement)
    CC->>CC: Handle acknowledgement
```

### ICA Channel Reactivation

```mermaid
sequenceDiagram
    participant CC as Controller Chain
    participant R as Relayer
    participant HC as Host Chain

    Note over CC,HC: Existing ICA channel expires or closes
    CC->>CC: Channel closes or times out
    CC->>R: ChannelOpenInit(Order, ConnectionHops, PortIdentifier, ChannelIdentifier, Version)
    Note right of CC: Reusing the same PortIdentifier and ConnectionID
    R->>HC: ChannelOpenTry(Order, ConnectionHops, PortIdentifier, ChannelIdentifier, Version)
    HC->>HC: Verify PortIdentifier and ConnectionID match the previous active channel
    HC->>HC: Verify the channel is in CLOSED state
    HC->>HC: Verify the new channel has the same order and version as the previous channel
    HC-->>R: ChannelOpenTry(Version)
    R->>CC: ChannelOpenAck(PortIdentifier, ChannelIdentifier, CounterpartyVersion)
    CC->>CC: Verify the CounterpartyVersion matches the previous active channel
    CC->>CC: SetActiveChannelID(PortIdentifier, ConnectionID, ChannelIdentifier)
    CC-->>R: ChannelOpenAck
    R->>HC: ChannelOpenConfirm(PortIdentifier, ChannelIdentifier)
    HC->>HC: SetActiveChannelID(CounterpartyPortIdentifier, ConnectionID, ChannelIdentifier)
```

**Comment by michaelfig** [comment]
Will be implemented in #8896.
